src/scene/tools/vcAddSimpleCrossSection.cpp

```diff
@@ -0,0 +1,59 @@
+#include "vcAddSimpleCrossSection.h"
+
+#include "vcState.h"
+#include "vcStrings.h"
+#include "vcCrossSectionNode.h"
+
+#include "udStringUtil.h"
+
+#include "imgui.h"
+
+vcAddSimpleCrossSection vcAddSimpleCrossSection::m_instance;
+
+void vcAddSimpleCrossSection::SceneUI(vcState *pProgramState)
+{
+  if ((pProgramState->sceneExplorer.selectedItems.size() == 1) && (pProgramState->sceneExplorer.selectedItems[0].pItem->pUserData != nullptr))
+  {
+    vcSceneItem *pSceneItem = (vcSceneItem *)pProgramState->sceneExplorer.selectedItems[0].pItem->pUserData;
+    pSceneItem->HandleSceneEmbeddedUI(pProgramState);
+  }
+}
+
+void vcAddSimpleCrossSection::HandlePicking(vcState *pProgramState, vcRenderData & /*renderData*/, const vcRenderPickResult & /*pickResult*/)
+{
+  if (!pProgramState->pActiveViewport->isMouseOver)
+    return;
+
+  if (pProgramState->sceneExplorer.clickedItem.pItem != nullptr && udStrEqual(pProgramState->sceneExplorer.clickedItem.pItem->itemtypeStr, "XSlice"))
```

Comment:
`XSlice`? Shouldn't it be called `XSection`?

Reply:
r/7characterlimit

Reply:
Of course, I forgot the limit is really low.